pkg/controller/common/association/ca.go

```diff
@@ -68,10 +69,12 @@ func ReconcileCASecret(
 		Expected:   &expectedSecret,
 		Reconciled: &reconciledSecret,
 		NeedsUpdate: func() bool {
-			return !reflect.DeepEqual(expectedSecret.Data, reconciledSecret.Data)
+			return !reflect.DeepEqual(expectedSecret.Data, reconciledSecret.Data) ||
+				!maps.IsSubset(expectedSecret.Labels, reconciledSecret.Labels)
```

Comment:
https://github.com/elastic/cloud-on-k8s/pull/2707 should help.
I'll do a second pass in a few days for in-flight PRs that did not benefit from it so we can keep this PR as is.